realm/realm-library/src/objectServer/java/io/realm/SyncSession.java

```diff
@@ -94,20 +102,78 @@
     private static final byte STATE_VALUE_INACTIVE = 3;
     private static final byte STATE_VALUE_ERROR = 4;
 
+    // List of Java connection change listeners
+    private final CopyOnWriteArrayList<ConnectionListener> connectionListeners = new CopyOnWriteArrayList<>();
+
+    // Reference to the token representing the native listener for connection changes
+    // Only one native listener is used for all Java listeners
+    private long nativeConnectionListenerToken;
+
+    // represent different states as defined in SyncSession::PublicConnectionState 'sync_session.hpp'
+    // saved here instead of as constants in ConnectionState.java to enable static checking by JNI
+    static final byte CONNECTION_VALUE_DISCONNECTED = 0;
+    static final byte CONNECTION_VALUE_CONNECTING = 1;
+    static final byte CONNECTION_VALUE_CONNECTED = 2;
+
     private URI resolvedRealmURI;
 
+    /**
+     * Enum describing the states a SyncSession can be in. The initial state is
+     * {@link State#INACTIVE}.
+     * <p>
+     * A Realm will automatically synchronize data with the server if the session is either {@link State#ACTIVE}
+     * or {@link State#DYING} and {@link #isConnected()} returns {@code true}.
+     */
     public enum State {
+
+        /**
+         * This is the initial state. The session is closed. No data is being synchronized. The session
+         * will automatically transition to {@link #WAITING_FOR_ACCESS_TOKEN} when a Realm is opened.
+         */
+        INACTIVE(STATE_VALUE_INACTIVE),
+
+        /**
+         * The user is attempting to synchronize data but needs a valid access token to do so. Realm
+         * will either use a cached token or automatically try to acquire one based on the current
+         * users login. This requires a network connection.
+         * <p>
+         * Data cannot be synchronized in this state.
+         * <p>
+         * Once a valid token is acquired, the session will transition to {@link #ACTIVE}.
+         */
         WAITING_FOR_ACCESS_TOKEN(STATE_VALUE_WAITING_FOR_ACCESS_TOKEN),
+
+        /**
+         * The Realm is open and data will be synchronized between the device and the server
+         * if the underlying connection is {@link ConnectionState#CONNECTED}.
+         * <p>
+         * The session will remain in this state until either the current login expires or the Realm
+         * is closed. In the first case, the session will transition to {@link #WAITING_FOR_ACCESS_TOKEN},
+         * in the second case, it will become either {@link #DYING} or {@link #INACTIVE}.
+         */
         ACTIVE(STATE_VALUE_ACTIVE),
+
+        /**
+         * The Realm was closed, but still contains data that needs to be synchronized to the server.
+         * The session will attempt to upload all local data before going {@link #INACTIVE}.
+         */
         DYING(STATE_VALUE_DYING),
-        INACTIVE(STATE_VALUE_INACTIVE),
+
+        @Deprecated
```

Comment:
is it deprecated in OS?

Reply:
Yes, it hasn't been used for a while there. I removed it from OS, but removing it from the public API is a breaking change.